.github/workflows/register-buildpack.yml

```diff
@@ -0,0 +1,260 @@
+              const buff = Buffer.from(value, 'utf-8');
+              return buff.toString('base64');
+            }
+
+            const createOrUpdateFileContents = async (path, message, content, sha = '') => {
+              let params = {
+                owner: process.env.GITHUB_OWNER,
+                repo: 'buildpack-registry',
+                path,
+                message,
+                content,
+                committer: {
+                    name: process.env.GITHUB_OWNER,
+                    email: 'longoria.public@gmail.com'
+                },
+                author: {
+                    name: process.env.GITHUB_OWNER,
+                    email: 'longoria.public@gmail.com'
+                }
+              }
+              if (sha !== '') {
+                  params = {...params, ...{sha}};
+              }
+              try {
+                await github.repos.createOrUpdateFileContents(params)
+              } catch (error) {
+                console.error(error)
+                process.exit(1)
+              }
+            }
+
+            let indexPath = Path.join('index')
+            if (nameLength === 1) {
+                indexPath = Path.join(indexPath, '1')
+            } else if (nameLength === 2) {
+                indexPath = Path.join(indexPath, '2')
+            } else if (nameLength === 3) {
+                indexPath = Path.join(indexPath, '3', buildpackInfo.name.slice(0, 2))
+            } else if (nameLength > 3) {
+                indexPath = Path.join(indexPath, buildpackInfo.name.slice(0, 2), buildpackInfo.name.slice(2, 4))
+            } else {
+                console.error(`buildpack name cannot be empty`)
+                process.exit(1)
+            }
+
+            indexPath = Path.join(indexPath, `${buildpackInfo.ns}_${buildpackInfo.name}`)
+            try {
+              const {data} = await github.repos.getContent({
+                owner: process.env.GITHUB_OWNER,
+                path: indexPath,
+                repo: 'buildpack-registry'
+              })
+
+              let buff = new Buffer.from(data.content, 'base64')
+              let fileContent = buff.toString('utf-8').trimEnd()
+              fileContent = fileContent + "\n" + JSON.stringify(buildpackInfo)
+
+              await createOrUpdateFileContents(indexPath, context.payload.issue.title, toBase64(fileContent), data.sha)
+
+              } catch (error) {
+                if (error.status && error.status === 404) {
+                  await createOrUpdateFileContents(indexPath, context.payload.issue.title, toBase64(JSON.stringify(buildpackInfo)))
+                } else {
+                  console.error(error)
+                  process.exit(1)
+                }
+              }
+
+      - uses: actions/github-script@v2
+        name: Close Issue on Success
+        with:
+          github-token: ${{ secrets.BOT_TOKEN }}
+          script: |
+            try {
+              await github.issues.createLabel({
```

Comment:
Isn't it simpler to just preemptively create it, instead of making an extra call every time it runs?